### PR TITLE
NEXT-22868 - Modal in actions should be optional

### PR DESCRIPTION
**Pull request by thuanlv287**

### CODE REVIEW

File: guides/plugins/plugins/framework/flow/add-flow-builder-action.md [hunk 1]
```diff
@@ -586,6 +586,55 @@ Click on [Save action] and we will get the result as below screenshot.
 
 ![Flow Builder create tag result](../../../../../.gitbook/assets/flow-builder-tag-result.png)
 
+#### Custom configuration for action without the modal
+
+If you do not need a modal for the configuration, it could be added automatically in action.
+
+Images, your action is already in the action list after <strong>Step 1.</strong>
```
[the remark this answers is not in the document]
Comment (elkmod) [reply guides/plugins/plugins/framework/flow/add-flow-builder-action.md:593]: Could we have a link to step 1 here? `[#step-1-link](first step)`.

File: guides/plugins/plugins/framework/flow/add-flow-builder-action.md [hunk 2]
```diff
@@ -586,6 +586,55 @@ Click on [Save action] and we will get the result as below screenshot.
...
+
+            // handle for the rest of actions.
+        },
+    },
+});
+```
+
+{% endcode %}
+
+After you click on the action, the new sequence will automatically add to the action list like this:
```
Comment (elkmod) [inline guides/plugins/plugins/framework/flow/add-flow-builder-action.md:634]:
```suggestion
Now, after you click on the action, the new sequence will automatically be added to the action list like this:
```

File: guides/plugins/plugins/framework/flow/add-flow-builder-action.md [hunk 3]
```diff
@@ -586,6 +586,55 @@ Click on [Save action] and we will get the result as below screenshot.
...
+
+{% endcode %}
+
+Now, after you click on the action, the new sequence will automatically be added to the action list like this:
+
+![Flow Builder create tag result](../../../../../.gitbook/assets/flow-builder-tag-result.png)
+
 ### Demo
 
 Now you can view the whole demo for this custom Flow builder trigger and action as below.
```
Comment (sushmangupta) [inline guides/plugins/plugins/framework/flow/add-flow-builder-action.md:640]:
```suggestion
You can view the whole demo for this custom Flow Builder trigger and action below:
```

**Review by sushmangupta — CHANGES_REQUESTED** [review]

**Review by sushmangupta — APPROVED** [review]

File: guides/plugins/plugins/framework/flow/add-flow-builder-action.md [hunk 4]
```diff
@@ -586,9 +586,58 @@ Click on [Save action] and we will get the result as below screenshot.
 
 ![Flow Builder create tag result](../../../../../.gitbook/assets/flow-builder-tag-result.png)
 
+#### Custom configuration for action without the modal
+
+You don't need a modal for the configuration. It can be added automatically.
+
+Imagine, your action is already in the action list after [the first step](https://developer.shopware.com/docs/guides/plugins/plugins/framework/flow/add-flow-builder-action#step-1-show-action-label-in-action-list)
```
Comment (elkmod) [inline guides/plugins/plugins/framework/flow/add-flow-builder-action.md:593]: Please use a relative link here.

**Review by elkmod — APPROVED** [review]

**Pull request merged**